lib/semian/sysv_shared_memory.rb

```diff
@@ -0,0 +1,50 @@
+module Semian
+  module SysVSharedMemory #:nodoc:
+    def self.included(base)
+      def base.do_with_sync(*names)
+        names.each do |name|
+          new_name = "#{name}_inner".freeze.to_sym
+          alias_method new_name, name
+          private new_name
+          define_method(name) do |*args, &block|
+            synchronize do
+              method(new_name).call(*args, &block)
+            end
+          end
+        end
+      end
+    end
```

Comment:
Yeah but how do you make sure the `SysV::Integer#value` is called, if all you do is `Simple::Integer#value`? The thing is you can't reuse `@value` at all since it doesn't exist in `SysV::Integer`. 

The C version and the Ruby version has two different ways of storing the value, and their sole accessor is through their `#value`. You can't both override it and call the original, i.e. you can't have your cake and eat it too.